doc/man7/migration_guide.pod

```diff
@@ -0,0 +1,1791 @@
+specified if required. The '-provider_path' must be specified before the
+'-provider' option. 
+
+'-providers' used by 'list', which will display the list of loaded providers,
+their names, version and status.  It optionally displays their gettable parameters.
+
+'-objects' used by 'list' which will display the list of built in objects,
+i.e. OIDs with names.
+
+`-crl_lastupdate` and `-crl_nextupdate` used by 'ca' allows explicit setting of
+fields in the generated CRL.
+
+=head3 Removed options
+
+Interactive mode is not longer available.
+
+The `-crypt` option used by `passwd`.
+The '-C' option used by `x509`, `dhparam`, `dsaparam`, and `ecparam`.
+
+=head3 Other Changes
+
+The output of Command line applications may have minor changes.
+These are primarily changes in capitalisation and white space.  However, in some
+cases, there are additional differences.
+For example, the DH parameters output from `dhparam` now lists 'P', 'Q', 'G' and
+'pcounter' instead of 'prime', 'generator', 'subgroup order' and 'counter'
+respectively.
+
+The openssl commands that read keys, certificates, and CRLs now
+automatically detect the PEM or DER format of the input files so it is not
+necessary to explicitly specify the input format anymore. However if the
+input format option is used the specified format will be required.
+
+`speed` no longer uses low-level API calls.
+This implies some of the performance numbers might not be fully comparable
+with the previous releases due to higher overhead. This applies particularly to
+measuring performance on smaller data chunks.
+
+'dhparam', 'dsa', 'gendsa', 'dsaparam', 'genrsa' and 'rsa' have been
+modified to use PKEY APIs.
+'genrsa' and 'rsa' now write PKCS #8 keys by default.
+
+=head3 Default settings
+
+"SHA256" is now the default digest for TS query used by`ts`.
+
+=head3 Deprecated apps
+
+'rsautl' has been deprecated, use 'pkeyutl' instead.
+'dhparam', 'dsa', 'gendsa', 'dsaparam', 'genrsa', 'rsa', 'genrsa' and 'rsa' are
+now in maintenance mode and no new features will be added to them.
+
+=head2 TLS Changes
+
+=over 4
+
+=item TLS 1.3 FFDHE key exchange support added
+
+This uses DH safe prime named groups.
+
+=item Support for fully "pluggable" TLSv1.3 groups.
+
+This means that providers may supply their own group implementations (using
+either the "key exchange" or the "key encapsulation" methods) which will
+automatically be detected and used by libssl.
+
+=item SSL_set1_host() and SSL_add1_host() Changes
+
+These functions now take IP literal addresses as well as actual hostnames.
+
+=item Added SSL option SSL_OP_CLEANSE_PLAINTEXT
+
+If the option is set, openssl cleanses (zeroize) plaintext bytes from
+internal buffers after delivering them to the application. Note,
+the application is still responsible for cleansing other copies
+(e.g.: data received by L<SSL_read(3)>).
+
+=item Secure renegotiation is now required by default for TLS connections
```

Comment:
Probably need an item for the recent change to disable client initiated reneg by default on the server

Reply:
Added..